crates/graphql-federated-graph/src/from_sdl.rs

```diff
@@ -1925,3 +1980,179 @@ fn test_missing_type() {
+      EXECUTION
+    }
+
+    type Query @join__type(graph: A) @join__type(graph: B) {
+      users: [User!]! @join__field(graph: A)
+      accounts: [Account!]! @join__field(graph: B)
+    }
+
+    type User @join__type(graph: A) @join__type(graph: B, key: "id") {
+      id: ID @join__field(graph: A, type: "ID") @join__field(graph: B, type: "ID!")
+      name: String @join__field(graph: B)
+      similarAccounts: [Account!]! @join__field(graph: B)
+    }
+    "###;
+
+    let expected = expect![[r#"
+        directive @core(feature: String!) repeatable on SCHEMA
+
+        directive @join__owner(graph: join__Graph!) on OBJECT
+
+        directive @join__type(
+            graph: join__Graph!
+            key: String!
+            resolvable: Boolean = true
+        ) repeatable on OBJECT | INTERFACE
+
+        directive @join__field(
+            graph: join__Graph
+            requires: String
+            provides: String
+        ) on FIELD_DEFINITION
+
+        directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+
+        directive @join__implements(graph: join__Graph!, interface: String!) repeatable on OBJECT | INTERFACE
+
+        directive @join__unionMember(graph: join__Graph!, member: String!) repeatable on UNION
+
+        enum join__Graph {
+            A @join__graph(name: "a", url: "http://localhost:4200/child-type-mismatch/a")
+            B @join__graph(name: "b", url: "http://localhost:4200/child-type-mismatch/b")
+        }
+
+        scalar join__FieldSet
+
+        scalar link__Import
+
+        type Admin
+            @join__type(graph: B)
+        {
+            id: ID @join__field(graph: B)
+            name: String @join__field(graph: B)
+            similarAccounts: [Account!]! @join__field(graph: B)
+        }
+
+        type Query
+            @join__type(graph: A)
+            @join__type(graph: B)
+        {
+            users: [User!]! @join__field(graph: A)
+            accounts: [Account!]! @join__field(graph: B)
+        }
+
+        type User
+            @join__type(graph: A)
+            @join__type(graph: B, key: "id")
+        {
+            id: ID @join__field(graph: A) @join__field(graph: B) @join__field(graph: B, type: "ID!")
+            name: String @join__field(graph: B)
+            similarAccounts: [Account!]! @join__field(graph: B)
+        }
+
+        enum link__Purpose {
+            """
+            `SECURITY` features provide metadata necessary to securely resolve fields.
+            """
+            SECURITY
+            """
+            `EXECUTION` features provide metadata necessary for operation execution.
+            """
+            EXECUTION
+        }
+
+        union Account
+            @join__unionMember(graph: B, member: "Admin")
+            @join__unionMember(graph: B, member: "User")
+         = User | Admin
+    "#]];
+
+    let actual = crate::render_sdl::render_federated_sdl(&super::from_sdl(sdl).unwrap()).unwrap();
+
+    expected.assert_eq(&actual);
```

Comment:
This looks like (effectively) a roundtrip test - would it be better just written as one?  Like just `input == render(from_sdl(input))`

Reply:
yeah should you're right, copy-pasted the input from rover so it's not exactly the same, should use the output

Reply:
well fixing this in a later PR